integration/mediation-tests/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/mediation/cache/Sample420TestCase.java

```diff
@@ -36,39 +36,55 @@
     @BeforeClass(alwaysRun = true)
     public void deployArtifacts() throws Exception {
         super.init();
-       // loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/cache/synapse_sample_420.xml");
         loadSampleESBConfiguration(420);
 
     }
 
     @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
-    @Test(groups = {"wso2.esb"}, description = "Creating simple cache sample 420 Test Case", enabled = false)
-    public void testSimpleCashing() throws AxisFault, XPathExpressionException, InterruptedException {
+    @Test(groups = {"wso2.esb"}, description = "Creating simple cache sample 420 Test Case")
+    public void testSimpleCachingNotExists() throws AxisFault, XPathExpressionException, InterruptedException {
         OMElement response;
 
-            long currTime = System.currentTimeMillis();
-            long timeDiff = 0;
+        long currTime = System.currentTimeMillis();
+        long timeDiff = 0;
 
-            response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL() , "", "IBM");
-            String firstResponse = response.getFirstElement().toString();
+        response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL(), "", "IBM");
+        String firstResponse = response.getFirstElement().toString();
 
-            while (timeDiff < 20000) {
+        while (timeDiff < 20000) {
 
-                response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL() , "", "IBM");
+            response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL(), "", "IBM");
 
-                if ( ! firstResponse.equalsIgnoreCase( response.getFirstElement().toString() ))
-                    assertFalse(false, "Caching is less than 20 seconds");
-
-                Thread.sleep(2000);
-                timeDiff = System.currentTimeMillis() - currTime;
+            if (!firstResponse.equalsIgnoreCase(response.getFirstElement().toString())) {
+                assertFalse(false, "Caching is less than 20 seconds");
             }
 
-            response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL() , "", "IBM");
+            Thread.sleep(2000);
+            timeDiff = System.currentTimeMillis() - currTime;
+        }
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL(), "", "IBM");
+
+        if (!firstResponse.equalsIgnoreCase(response.getFirstElement().toString())) {
+            assertTrue(true, "Caching is more than 20 seconds");
```

Comment:
This will not fail either. Shall we correct in all places?

Reply:
Fixed in #983